Add docs about upgrading primary mesh gateways (#15032)
hashicorp/consul: website/content/docs/connect/gateways/mesh-gateway/wan-federation-via-mesh-gateways.mdx

- Ensure any API request that activates datacenter request forwarding. such as
  [`/v1/catalog/services?dc=<OTHER_DATACENTER_NAME>`](/api-docs/catalog#dc-1)
  succeeds.

### Upgrading the primary gateways

Once federation has been established, updates for the addresses of new primary

> gateways are propagated through the gateways in the primary themselves. If the primary

Author: No it's not, the updates that need to be propagated are address changes for mesh gateways in the primary datacenter. But your question helped me think of a different way to clarify this.

I pushed an update to make explicit that this propagation is actually initiated by the secondary datacenters. 

Here's an example that helped me think through this:
1. Primary datacenter has three mesh gateways: [1,2,3]
2. Secondary datacenter is continuously dialing the primary datacenter via gateways [1,2,3] to receive updated primary gateway addresses.
3. Primary datacenter quickly swaps gateways [1,2,3] for [4,5,6], taking down [1,2,3].

If step 3 happens fast enough, then the queries from the secondary datacenter in step 2 do not return addresses of [4,5,6] before [1,2,3] become unavailable. Step 2 will continue to dial [1,2,3] because those are the only addresses it knows about, but because they're gone that means the primary datacenter is unreachable.

- Ensure any API request that activates datacenter request forwarding. such as
  [`/v1/catalog/services?dc=<OTHER_DATACENTER_NAME>`](/api-docs/catalog#dc-1)
  succeeds.

### Upgrading the primary gateways

Once federation is established, secondary datacenters will continuously request

> updated mesh gateway addresses from the primary datacenter. These requests
> themselves flow through the mesh gateways of the primary datacenter, since
> secondary datacenters cannot dial the primary datacenter's Consul servers directly.

Reviewer: ```suggestion
updated mesh gateway addresses from the primary datacenter. Consul route the requests
 through the primary datacenter's mesh gateways. This is because
secondary datacenters cannot directly dial the primary datacenter's Consul servers.
```
Trying to make the language more precise by specifically attributing the action to Consul ("Consul routes requests . . .") and putting the adverb directly in front of the verb it's modifies ("directly dial"). Ignore if you don't feel these are improvements.

Author: Looks good, will apply it.
